go/tasks/plugins/array/awsbatch/monitor.go

```diff
@@ -108,6 +109,10 @@ func CheckSubTasksState(ctx context.Context, taskMeta core.TaskExecutionMetadata
 			} else {
 				msg.Collect(childIdx, "Job failed")
 			}
+
+			if subJob.Status.Phase == core.PhaseRetryableFailure && retryLimit == int64(len(subJob.Attempts)) {
+				totalRetryLimitExceeded++
+			}
```

Comment:
Does it make sense to transition the subtask to PermanentFailure here (like we do in the [array plugin](https://github.com/flyteorg/flyteplugins/blob/d95b86b4ba9e6fcab18b3ea834047c4248cf314a/go/tasks/plugins/array/k8s/management.go#L251-L255)) rather than tracking the number of retryable failures which exceeded the retry limit?

Reply:
updated it